src/Streamly/Internal/LZ4.hs

```diff
@@ -242,7 +242,7 @@ compressChunk i ctx arr = do
               let dstEnd = dstBegin `plusPtr`
                             ((fromIntegral :: CInt -> Int) compLen + 8)
                   compArr = MArray.Array fptr dstEnd dstMax
-              Array.unsafeFreeze <$> MArray.shrinkToFit compArr
+              return $ Array.unsafeFreeze compArr
```

Comment:
It should be safe to shrink here. As we depend on the previous uncompressed array and not the compressed one.

Reply:
I'm adding this as a comment just to make it extra clear.